### PR TITLE
[10.0][ADD] purchase_comment_template

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Purchase Comments
=================

Adds comments on

1. purchases orders:

   The comments can be edited directly on the purchases orders or loaded from
   templates.

   Two positions are available for the comments:

   - above purchase order lines
   - below purchase order lines

2. purchase order lines:

   The formatted comments can be edited directly on the purchases order lines.

### CODE REVIEW

**Review by AaronHForgeFlow — COMMENTED** [review]
Hi, thank you for this PR. Tested functionality and it works fine. Is it possible to pass the comments to the supplier invoice as it is done in the sale module https://github.com/OCA/sale-reporting/blob/10.0/sale_comment_template?

**Comment by MiquelRForgeFlow** [comment]
No.

**Comment by AaronHForgeFlow** [comment]
It would be good to have the same capabilities in purchases than in sales. Otherwise, why adding this module instead of migrating this one? https://github.com/OCA/purchase-reporting/pull/21

**Comment by MiquelRForgeFlow** [comment]
Because `sale_comment_template` is already merged, so I think it is better to do a `purchase_comment_template` instead of a `purchase_order_report_with_note`. Just consistency. And I don't bother about the "comments passed to invoice" feature.

**Comment by AaronHForgeFlow** [comment]
ok

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
:face_with_head_bandage: 

`Element '<xpath expr="//table[@class='table table-condensed']/tbody/tr/td[1]">' cannot be located in parent view`

Update: This module does not break by itself. But it makes others to fail, for example, purchase_order_line_sequence (https://github.com/OCA/purchase-workflow/tree/10.0/purchase_order_line_sequence)

Please do not use replace in the view.

File: purchase_comment_template/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ "author": "Eficent, Odoo Community Association (OCA)",
+ "license": "AGPL-3",
+ "depends": ["purchase",
+             "invoice_comment_template",
+             ],
+ "data": ["views/purchase_order_view.xml",
+          'views/base_comment_template_view.xml',
+          'views/report_purchaseorder.xml',
+          'views/report_quotation.xml',
+          'security/ir.model.access.csv',
```
Comment (nikul-serpentcs) [inline purchase_comment_template/__manifest__.py]: @mreficent IMO `security` should be First

File: purchase_comment_template/views/report_purchaseorder.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      inherit_id="purchase.report_purchaseorder_document">
+      <xpath expr="//div[@class='oe_structure']" position="before">
+        <template id="purchase_css" inherit_id="web.layout">
+            <xpath expr="//head" position="inside">
+                <link rel="stylesheet"
+                    href="/purchase_comment_template/static/src/css/purchase.css" />
+            </xpath>
+        </template>
+      </xpath>
+      <xpath expr="//table[contains(@class, 'table-condensed')]" position="before">
```
Comment (nikul-serpentcs) [inline purchase_comment_template/views/report_purchaseorder.xml]: here use `hasclass`.

File: purchase_comment_template/views/report_quotation.xml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      inherit_id="purchase.report_purchasequotation_document">
+      <xpath expr="//div[@class='oe_structure']" position="before">
+        <template id="purchase_css" inherit_id="web.layout">
+            <xpath expr="//head" position="inside">
+                <link rel="stylesheet"
+                    href="/purchase_comment_template/static/src/css/purchase.css" />
+            </xpath>
+        </template>
+      </xpath>
+      <xpath expr="//table[contains(@class, 'table-condensed')]" position="before">
```
Comment (nikul-serpentcs) [inline purchase_comment_template/views/report_quotation.xml]: same here

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: purchase_comment_template/views/report_purchaseorder.xml (hunk 2 above)
Comment (MiquelRForgeFlow) [reply purchase_comment_template/views/report_purchaseorder.xml]: done

**Review by AaronHForgeFlow — APPROVED** [review]
LGTM :+1:

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM

**Pull request merged**